src/Parser/Mage/Fire/Modules/Items/MarqueeBindingsOfTheSunKing.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import ITEMS from 'common/ITEMS';
+import SPELLS from 'common/SPELLS';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import ItemDamageDone from 'Main/ItemDamageDone';
+import ItemLink from 'common/ItemLink';
+import { formatPercentage } from 'common/format';
+
+class MarqueeBindingsOfTheSunKing extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+  
+  damage = 0;
+  legendaryProcs = 0;
+  overwrittenProcs = 0;
+  buffedPyroblasts = 0;
+  expiredProcs = 0
+  
+  lastBuffTimestamp;
+  lastPyroCastTimestamp;
+  
+  on_initialized() {
+  this.active = this.combatants.selected.hasWrists(ITEMS.MARQUEE_BINDINGS_OF_THE_SUN_KING.id);
+  }
+
+	// get total count of procs
+  
+    on_toPlayer_applybuff(event) {
+	  if (event.ability.guid !== SPELLS.KAELTHAS_ULTIMATE_ABILITY.id) {
+	  return;
+    }
+  this.legendaryProcs += 1;
+  }
+  
+   // get count of refreshes
+  	on_byPlayer_refreshbuff(event) {
+    if (event.ability.guid !== SPELLS.KAELTHAS_ULTIMATE_ABILITY.id) {
+			return;
+		}
+		this.overwrittenProcs += 1;
+		this.legendaryProcs += 1;
+	}
+	
+	
+	// this section needs explaining:
+	// from looking through logs, the order of events is buff expires -> pyroblast cast begins within ~100ms -> pyroblast hits within 4.5 seconds
+	// i'm not sure this is the best way to accomplish this but it seems to work
+	
+	//get timestamp when buff expires
+	
+	on_byPlayer_removebuff(event) {
+    if (event.ability.guid !== SPELLS.KAELTHAS_ULTIMATE_ABILITY.id) {
+			return;
+		}
+		this.lastBuffTimestamp = this.owner.currentTimestamp;
+			}
+	
+	// for each pyroblast, check if buff expired within the past 100ms. if so, count it as buffed by the legendary
+	// additionally, set timestamp of cast to do a damage check
+	// i suppose it's possible for a player to instant cast pyroblast with hot streak within 100ms of the buff ending so it checks that this isn't the case.
+	
+	on_byPlayer_cast(event) {
+    if (event.ability.guid !== SPELLS.PYROBLAST.id) {
+			return;
+		}
+    if (this.lastBuffTimestamp > this.owner.currentTimestamp - 100 && !this.combatants.selected.hasBuff(SPELLS.HOT_STREAK) ) {
+			this.buffedPyroblasts += 1;
+			this.lastPyroCastTimestamp = this.owner.currentTimestamp;
+		}
+		}
+
+	// after being cast, pyroblast takes a few seconds to travel and hit the target.
+	// allows 3 seconds for the pyroblast to hit, then resets timestamp so additional instant pyroblasts within the time limit don't count.
+	// this is messy and i don't know if there's an easier way
+	
+	on_byPlayer_damage(event) {
+    if (event.ability.guid !== SPELLS.PYROBLAST.id) {
+			return;
+		}
+		
+	if (this.owner.currentTimestamp - 3000 < this.lastPyroCastTimestamp)	
+			this.damage += event.amount + (event.absorbed || 0);
+			this.lastPyroCastTimestamp = 0;
+			console.log(this.damage);
```

Comment:
console logs for non-error behavior shouldn't make it to production. Please either remove this or hide it behind a debug flag.